code/Forms/UsedOnTable.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace SilverStripe\Admin\Forms;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Forms\FormField;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\RecursivePublishable;
+use SilverStripe\Versioned\Versioned;
+
+/**
+ * Provides a table which displays the provided record's owners
+ *
+ * NOTE: This should be a dataless field but due to how field actions work - it needs recursive checking
+ */
+class UsedOnTable extends FormField
+{
+    protected $schemaDataType = FormField::SCHEMA_DATA_TYPE_CUSTOM;
+
+    /**
+     * @skipUpgrade
+     * @var string
+     */
+    protected $schemaComponent = 'UsedOnTable';
+
+    /**
+     * @config
+     * @var array
+     */
+    private static $allowed_actions = [
+        'usage'
+    ];
+
+    /**
+     * @var DataObject|RecursivePublishable
+     */
+    protected $record;
+
+    /**
+     * @param string $name
+     * @param DataObject|RecursivePublishable $record
+     */
+    public function __construct($name, $record)
```

Comment:
Mandatory record field will make this field impossible to scaffold without a parent record. Shift this to a setter instead.

See `FileUploadReceiver::getRecord()` for lazy-loading record from a parent form. It should be inferrable from `$this->getForm()->getRecord()` instead of requiring it upfront.